main.cpp

```diff
@@ -1010,13 +1014,366 @@ Part 1e - Step 19: Request a review
+    int clockSpeed = 2400;
+    int memoryBusSize = 1400;
+    int processorSpeed = 320;
+    int numberOfInputOutputStreams = 6;
+    int fanSpeed = 12.f;
+};
+// 3 things it can do:
+//     1) Increase ventilation amount
+    void fansOn(GPU gpuFans);
+
+//     2) Render graphics
+    void graphicsRendering(GPU gpu);
+
+//     3) Overlock processing speed
+    float processorOverlocking(int currentProcessorSpeed, int inputClockSpeed, GPU gpu); // NOTE: returns overlocking amount as a value.
+
+
+/*
+Thing 8) Memory
+5 properties:
+    1) capacity size (int)
+    2) storage type (std::string)
+    3) data rate (float)
+    4) seek time (double)
+    5) RAM type (std::string)
+3 things it can do:
+    1) write data
+    2) read data
+    3) erase data
+*/
+struct Memory
+{
+// 5 properties:
+    int capacitySize = 256;
+    std::string storageType = "SSD";
+    float dataRate = 18.5f;
+    double seekTime = 46.889;
+    std::string ramType = "DRAM";
+};
+// 3 things it can do:
+//     1) write data
+    int writeDataToMemory(Memory memoryDataSource); // NOTE: returns a value to signify successful writing to memory.
+
+//     2) read data
+    void openDataLocation(Memory memoryaddress);
+
+//     3) erase data
+    void deleteData(std::string selectedData);
+
+
+/*
+Thing 9) USB port
+5 properties:
+    1) data transfer speed (float)
+    2) number of ports (int)
+    3) connector type (std::string)
+    4) power output (int)
+    5) backward compatibility (std::string)
+3 things it can do:
+    1) connect external devices
+    2) transfer files
+    3) charge devices
+*/
+struct UsbPort
+{
+// 5 properties:
+    float dataTransferSpeed = 16.8f;
+    int numberOfPorts = 4;
+    std::string connectorType = "USB-C";
+    int powerOutput = 20;
+    std::string backwardCompatibility = "USB 3.0 and USB 2.0 compatible";
+};
+// 3 things it can do:
+//     1) connect external devices
+    int connectingDevice(UsbPort usbInput); // NOTE: returns a value indicating successful or failed connection.
+
+//     2) transfer files
+    void fileTransferring(std::string fileToSend);
+
+//     3) charge devices
+    bool chargingConnectedDevice(UsbPort usbConnected); // NOTE: returns a bool value to cause system to display positive LED charging light.
+
+
+/*
+Thing 10) Gaming Laptop
+5 properties:
+    1) screen.
+    2) CPU.
+    3) GPU.
+    4) Memory.
+    5) USB port.
+3 things it can do:
+    1) Load a game.
+    2) Connect peripherals.
+    3) Display image.
+*/
+struct GamingLaptop
+{
+// 5 properties:
+    Screen screen;
+    CPU cpu;
+    GPU gpu;
+    Memory memory;
+    UsbPort usbport;
+};
+// 3 things it can do:
+//     1) Load a game.
+    void loadingGame(std::string gameApplicationToOpen);
+
+//     2) Connect peripherals.
+    int connectingDevice(std::string deviceName); // NOTE: returns a value to signify successful or failed connections.
 
+//     3) Display image.
+    void displayingToScreen(GamingLaptop gamingLaptopDisplay);
```

Comment:
These have not been declared as member functions
they are outside the struct's closing `}`
- [ ] revise